packages/squiggle-lang/src/rescript/Reducer/Reducer_Peggy/Reducer_Peggy_GeneratedParser.peggy

```diff
@@ -139,16 +139,18 @@ power
 chainFunctionCall
   = head:unary tail:(_ ('->'/'|>') _nl chained:chainedFunction {return chained})* 
   { return tail.reduce(function(result, element) {
-      return h.makeFunctionCall(element.fnName, [result, ...element.args], location())
+      return h.makeFunctionCall(h.postOperatorToFunction['()'], [element.callable, result, ...element.args], location())
     }, head)}
 
   chainedFunction
-    = fn:variable '(' _nl args:array_functionArguments _nl ')' 
-      { return {fnName: fn.value, args: args}}
-      / fn:variable '(' _nl ')' 
-      { return {fnName: fn.value, args: []}}
-      / fn:variable
-      { return {fnName: fn.value, args: []}}
+    = fn:callableBasicValue '(' _nl args:array_functionArguments _nl ')' 
+      { return {callable: fn, args: args}}
+    / fn:callableBasicValue '(' _nl ')' 
+      { return {callable: fn, args: []}}
+    / fn:callableBasicValue
+      { return {callable: fn, args: []}}
+      
+  callableBasicValue = valueConstructor / variable   
```

Comment:
I don't want to overcomplicate this PR is this is hard to do (and it might be if arbitrary expressions are greedy and would eat the final `()`), but I see that this still doesn't allow arbitrary expressions on the right side of `->`.
Maybe it could be some wider rule than `valueConstructor / variable `?

Compare:
- https://squiggle-website-git-issue-655-ch-85ff2f-quantified-uncertainty.vercel.app/playground#code=eNqrVirOyC8PLs3NTSyqVLIqKSpN1QELuaZkluQXwUQy8zJLMhNzggtLM9PTc1KDS4oy89KVrJRMde2q06yqaypqKrQNa2v10pRqAfYqHhA%3D - doesn't parse
- https://rescript-lang.org/try?code=DYUwLgBAXhC8EFYC0A+A3gIgGYYFwQA84VCBqARgF8BtbDAXSA - parses correctly

Reply:
I will add post expressions (".", "[]") to the mix. 
- Just arbitrary expressions cannot be chained into because they would eat the final "()".  
- Also binary expressions should not be chained into because they always result in logic, numerical or distribution results.

Reply:
```
   describe("to expression", () => {
    testEvalToBe("1->{f:{|x|x+2}}.f", "Ok(3)")
    testEvalToBe("1->{f:{|x|x+2}}['f']", "Ok(3)")
   })

```